apps/prairielearn/assets/scripts/instructorAssessmentInstancesClient.tsx

```diff
@@ -401,93 +414,245 @@ onDocumentReady(() => {
-        </p>
-        ${row.has_closed_instance
-          ? html`
-              <div class="form-check mb-2 reopen-closed-field">
+  function TimeLimitExplanation({ plusMinus }: { plusMinus: TimeLimitPlusMinus }) {
+    let explanation = '';
+    if (plusMinus === 'set_total') {
+      explanation =
+        'Updating the total time limit will set the given amount of time for the assessment based on when the assessment was started.';
+    } else if (plusMinus === 'set_rem') {
+      explanation =
+        'Updating the time remaining will set the given amount of time for the assessment based on the current time.';
+    } else if (plusMinus === 'set_exact') {
+      explanation = 'This will set the exact closing time for the assessment.';
+    } else if (plusMinus === '+1') {
+      explanation = 'This will add the given amount of time to the remaining time limit.';
+    } else if (plusMinus === '-1') {
+      explanation = 'This will subtract the given amount of time from the remaining time limit.';
+    } else if (plusMinus === 'unlimited') {
+      explanation = 'This will remove the time limit and the assessment will remain open.';
+    } else if (plusMinus === 'expire') {
+      explanation = 'This will expire the time limit and the assessment will be closed.';
+    }
+    return (
+      <small class="text-muted d-block" style="width:400px; max-width:90vw">
+        {explanation}
+      </small>
+    );
+  }
+
+  function TimeLimitEditPopover({
+    row,
+  }: {
+    row: {
+      action: string;
+      assessment_instance_id: number;
+      date: string;
+      has_closed_instance: boolean;
+      has_open_instance: boolean;
+      total_time: string;
+      time_remaining: string;
+      time_remaining_sec: number;
+      open: boolean;
+    };
+  }) {
+    // Because we need to update values on the Preact virtual DOM, we need to manage our
+    // form values as state. These can then be passed and used by the child components to
+    // display values or used for conditional rendering.
+    const [form, setForm] = useState<{
+      plus_minus: TimeLimitPlusMinus;
+      time_add: number;
+      date: string;
+      reopen_closed: boolean;
+      reopen_without_limit: boolean;
+    }>({
+      plus_minus: '+1',
+      time_add: 5,
+      date: Temporal.Now.zonedDateTimeISO(timezone).toPlainDateTime().toString().slice(0, 16),
+      reopen_closed: false,
+      reopen_without_limit: true,
+    });
+    const showTimeLimitOptions =
+      row.action === 'set_time_limit_all' || row.open || !form.reopen_without_limit;
+
+    function updateFormState<T extends keyof typeof form>(key: T, value: (typeof form)[T]) {
+      setForm({
+        ...form,
+        [key]: value,
+      });
+    }
+
+    function proposedClosingTime() {
+      const startDate = row.date;
+      let totalTime = 0;
+      if (row.total_time !== 'Open (no time limit)' && row.total_time !== 'Closed') {
+        totalTime = parseFloat(row.total_time);
+      }
```

Comment:
😬 surely it'd be better to use `total_time_sec` than trying to parse the human-readable string here? If you're not sure what properties are available on `row`, check out the request/response for `raw_data.json` in your browser's dev tools, or just log out `row` here.

Reply:
Yes, now using `total_time_sec` in [42543c9](https://github.com/PrairieLearn/PrairieLearn/pull/10800/commits/42543c9a1ca769d8b9f72a66c1f6d1916060f477).